### PR TITLE
Add functionality for rebasebot to rebase onto viable/strict branch

### PR DESCRIPTION
**Pull request by swang392**
Fixes #325
**Summary**: Currently, the pytorchbot only allows for rebasing to the master branch. These modifications add functionality for rebasing to the 'viable/strict' branch of pytorch/pytorch by adding a flag to the comment.
**Test Plan:** tested manually on personal fork ([#1](https://github.com/swang392/pytorch/pull/1)), and included a test case in test_tryrebase.py that checks if rebasing to viable/strict branch was successful.

### CODE REVIEW

File: .github/scripts/test_tryrebase.py [hunk 1]
```diff
@@ -22,6 +22,21 @@ def test_rebase(self, mocked_post_comment: Any, mocked_run_git: Any, mocked_gql:
         mocked_run_git.assert_has_calls(calls)
         self.assertTrue("Successfully rebased `master` onto `master`" in mocked_post_comment.call_args[0][3])
 
+    @mock.patch('trymerge.gh_graphql', side_effect=mocked_gh_graphql)
+    @mock.patch('gitutils.GitRepo._run_git')
+    @mock.patch('tryrebase.gh_post_comment')
+    def test_rebase_to_stable(self, mocked_post_comment: Any, mocked_run_git: Any, mocked_gql: Any) -> None:
+        "Tests rebase to viable/strict successfully"
+        pr = GitHubPR("pytorch", "pytorch", 77907)
```
Comment (janeyx99) [inline .github/scripts/test_tryrebase.py:30]: Can we use an older PR so we don't need to add to gql_mocks.json lol

Comment (janeyx99) [reply .github/scripts/test_tryrebase.py:30]: aka one that already existed (like would 31093 work?)

File: .github/scripts/tryrebase.py [hunk 2]
```diff
@@ -13,13 +13,17 @@ def parse_args() -> Any:
...
 
 
-def rebase_onto(pr: GitHubPR, repo: GitRepo, dry_run: bool = False) -> None:
+def rebase_onto(pr: GitHubPR, repo: GitRepo, dry_run: bool = False, stable: bool = False) -> None:
     branch = f"pull/{pr.pr_num}/head"
-    onto_branch = pr.default_branch()
+    if stable:
+        onto_branch = "viable/strict"
+    else:
+        onto_branch = pr.default_branch()
```
Comment (janeyx99) [inline .github/scripts/tryrebase.py:26]:
```suggestion
    onto_branch = "viable/strict" if stable else pr.default_branch()
```

File: .github/scripts/tryrebase.py [hunk 3]
```diff
@@ -39,11 +43,14 @@ def rebase_onto(pr: GitHubPR, repo: GitRepo, dry_run: bool = False) -> None:
                         "git pull --rebase`)", dry_run=dry_run)
 
 
-def rebase_ghstack_onto(pr: GitHubPR, repo: GitRepo, dry_run: bool = False) -> None:
+def rebase_ghstack_onto(pr: GitHubPR, repo: GitRepo, dry_run: bool = False, stable: bool = False) -> None:
     if subprocess.run([sys.executable, "-m", "ghstack", "--help"], capture_output=True).returncode != 0:
         subprocess.run([sys.executable, "-m", "pip", "install", "ghstack"])
     orig_ref = f"{re.sub(r'/head$', '/orig', pr.head_ref())}"
-    onto_branch = pr.default_branch()
+    if stable:
```
Comment (janeyx99) [inline .github/scripts/tryrebase.py:50]: same sugg as above

File: .github/scripts/test_tryrebase.py (hunk 1 above)
Comment (swang392) [reply .github/scripts/test_tryrebase.py]: changed the PR to 31093, but it still says that gql_mocks.json was changed

Comment (swang392) [reply .github/scripts/test_tryrebase.py]: nvm fixed!

**Review by clee2000 — APPROVED** [review]
lgtm

**Review by janeyx99 — APPROVED** [review]

**Pull request closed without merging**